File: 63_Name/java/Name.java
import java.util.Arrays;
import java.util.Scanner;

public class Name {

    public static void printempty() { System.out.println(" "); }

    public static void print(String toprint) { System.out.println(toprint); }

    public static void main(String[] args) {
        print("                                          NAME");
        print("                         CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY");
        printempty();
        printempty();
        print("HELLO.");
        print("MY NAME iS CREATIVE COMPUTER.");
        print("WHATS YOUR NAME? (FIRST AND LAST)");

        Scanner namesc = new Scanner(System.in);
        String name = namesc.nextLine();

        String namereversed = new StringBuilder(name).reverse().toString();

        char namesorted[] = name.toCharArray();
        Arrays.sort(namesorted);

        printempty();
        print("THANK YOU, " + namereversed);
        printempty();
        print("OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART");
        print("COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!");
        printempty();
        printempty();
        print("BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER.");

        print("LET'S PUT THEM IN ORDER LIKE THIS: " + new String(namesorted));
        printempty();
        printempty();

        print("DON'T YOU LIKE THAT BETTER?");
        printempty();

        Scanner agreementsc = new Scanner(System.in);
        String agreement = agreementsc.nextLine();

        if (agreement.equalsIgnoreCase("yes")) {
            print("I KNEW YOU'D AGREE!!");
        } else {
            print("I'M SORRY YOU DON'T LIKE IT THAT WAY.");
            printempty();
            print("I REALLY ENJOYED MEETING YOU, " + name);
            print("HAVE A NICE DAY!");
        }
    }
}

File: 63_Name/lua/name.lua
-- HELLO
--
-- Converted from BASIC to Lua by Recanman

local function tab(space)
    local str = ""

    for _ = space, 1, -1 do
        str = str .. " "
    end

    return str
end

-- reused from Bagels.lua
function getInput(prompt)
    io.write(prompt)
    io.flush()
    local input = io.read("l")
    if not input then  --- test for EOF
        print("GOODBYE")
        os.exit(0)
    end
    return input
end

print(tab(33) .. "HELLO\n")
print(tab(15) .. "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n")
print("\n")
print("\n")
print("\n")

print("HELLO.  MY NAME IS CREATIVE COMPUTER.\n")
print("\n")
print("\n")

print("WHAT'S YOUR NAME (FIRST AND LAST)")

local ns = getInput("? ")
local l = string.len(ns)
print("\n")

local function main()
    print("THANK YOU, " .. string.reverse(ns) .. ".\n")

    print("OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART")
    print("COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!\n")
    print("BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER.\n")
    print("LET'S PUT THEM IN ORDER LIKE THIS: ")
    
    local b = {}
    
    for i = 1, l, 1 do
        local letter = string.sub(ns, i, i)
        b[i] = string.byte(letter)
    end
    
    table.sort(b, function(v1, v2)
        return v1 < v2
    end)
    
    local str = ""
    for _, letter in ipairs(b) do
        str = str .. string.char(letter)
    end
    
    str = string.reverse(str)
    print(str)
    
    print("\n\n")
    print("DON'T YOU LIKE THAT BETTER")
    
    local ds = getInput("? ")
        
    if ds == "YES" then
        print("I KNEW YOU'D AGREE!!\n")
    else
        print("I'M SORRY YOU DON'T LIKE IT THAT WAY.\n")
    end
    
    print("I REALLY ENJOYED MEETING YOU " ..  ns .. ".\n")
    print("HAVE A NICE DAY!\n")
end

main()
